### DISCUSSION TITLE
Lan support on H3 (Kopp Branding)

### DISCUSSION THREAD

**Discussion by sleepymaxx** [nathanmarlor/foxess_modbus#466] [Support]
Hallo alltogether, sorry again the noob has repeated questions... :O, 

i have set up lan connections... but im failing so please help me again.

i know my ips. they are set up right (ping in command is functioning very well)
i have shut down and powerd up my inverter
the port is set to 502 (and portscan is finding this as an open port)
![grafik](https://github.com/nathanmarlor/foxess_modbus/assets/103333137/d19cbbfd-9a21-4865-aa1b-632357819f6c)
the inverter id is set to 247
im on foxess_modbus version 1.8.0b4
im on newest HomeAssistant Core 2023.11.2
                    Supervisor 2023.11.0
                    Operating System 11.1
                    Frontend 20231030.2
Software Versions on My FoxESS Master:  Manager    1.59   (my master and Slave are brandet versions of FoxESS)
                                                              Slave         1.03   (they are named Kopp Kuara 10.0-3-H)
                                                              Master      1.69    (in reality they are FoxESS H3-10.0-E)
                                                 Slave:     Manager   1.59
                                                              Slave         1.03
                                                              Master      1.69
(yes the software version is very old but i have only updated them twice since having the inverters +++ beause all following versions had very much problems in the consuption of power while in idle state... and im awating the upcoming versions in oder to get some with also good idle consuption.... and these version formerly had a good funcion over lan....)

While setting up the integration im getting the following error: 
![Screenshot 2023-11-13 215346](https://github.com/nathanmarlor/foxess_modbus/assets/103333137/47657859-512d-4639-a5b1-e0a85cb9742a)

Sorry for bad english and tooo much text... 
PS: yes im aware that the lan integration is powerless and only good for reading not for input and so on, but anyway my only goal is here to get the data out of the inverter and nothing else...  because im needing the modbus connectors for some other use.... : the 22 kW Wallbox...

**Comment by canton7** [comment]
The H3s don't have a lan port. Are you sure your Kopp branded H3 has a lan port which it exposes modbus over?

**Reply by sleepymaxx** [reply]
Yes .... you really drove me nuts.... had to go downstairs to controll .. .besides the usb-wlan Dongle towards the battery connectors  there is a ethernet port labeled LAN .... and yes it is NOT the BMS- Lan Connector (nor the master - slave Lan port)

**Reply by canton7** [reply]
I'm not sure then. It looks like Kopp have done something custom - it's not on the H3, and although it seems like it's probably taking modbus, it looks like it isn't exposing the same registers as the H3 exposes over RS485.

I thought you connected over RS485 in #431. Are you now changing to use a direct lan connection?

**Reply by sleepymaxx** [reply]
you are right.. my RS485 Connetion is working very well as you predicted .... the last days i worked on getting the overall Energy which is after my Heatpump exported to the grid into my Hot Water reservoir...  #[https://community.home-assistant.io/t/mypv-integation-started/395584/28](url)

so..... is my plan - so i can use the now for modbus use pins ---use for the connection to my (will in future be installed 22 kWh Wallbox... ) at the moment its not so urgent because the mini cars in germany are bullshit - and for the big ones im way toooo stingy .... (in Bavaria swabia this is a very good behaviour... ;-) )

**Reply by sleepymaxx** [reply]
Do you have any ideas of portscanning or anything else i can try / use to get the modubs readings out... ? so i will test anything to get this pushed further...

**Reply by canton7** [reply]
All I can say is that it doesn't look like it's exposing the normal registers over the ethernet connection. You can try reading various registers with a tool like modpoll, see if you can work out what it does expose. But:
1. My guess is that the ethernet connection is being used for something completely different and doesn't provide access to PV, load, feed-in, etc etc. Why would they provide the same registers but just change the addresses? It doesn't make sense.
2. Even if the ethernet does provide the same registers as RS485 but at different addresses, you'll have to map out all of those addresses. For the H3 that took several people quite a long time.

**Reply by sleepymaxx** [reply]
so i see... Thank you again very much .... and i´m at least clever enought so see that i will not have the time nor the knowlege do do/learn this by myself... so i will lay this (for today down... ) at least i won´t give up... knowing myslef too good ... so in future there will other posibilities... 
Thank´s again...
